### PR TITLE
Support Nginx monitoring.

### LINKED ISSUE #11534: [Feature] Support Nginx monitoring.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Nginx is a popular HTTP and reverse proxy server. I was already doing this work last weekend, please assign this task to me, more details and progress I will sync here.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by weixiang1862**
### Nginx monitoring
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If it's UI related, attach the screenshots below.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #11534.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

50cdbaf31624c4c6442e35c2a2eebe8
4c01ad9c80ca6e6bae35b19bef7cb65
a24f6b86a30843bea2d93d932fa8399
0f893b60d8b99f9537db61c7679c9c9

### CODE REVIEW

**Comment by wu-sheng** [comment]
From the screenshot, the log level is absent.

File: test/e2e-v2/cases/nginx/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - ./nginx.conf:/var/nginx/conf.d/nginx.conf
+      - /tmp/skywalking-logs/nginx:/var/log/nginx
+    entrypoint: ['bash', '-c', 'sleep 5 && openresty -c /var/nginx/conf.d/nginx.conf']
+    depends_on:
+      oap:
+        condition: service_healthy
+    ports:
+      - 8080
+    expose:
+      - 8080
```
Comment (wankai123) [inline test/e2e-v2/cases/nginx/docker-compose.yml:42]:
```suggestion
```

File: test/e2e-v2/cases/nginx/e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      expected: expected/metrics-has-value-label.yml
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics multiple-linear --name=meter_nginx_service_http_status --service-name=nginx::e2e-test | yq e 'to_entries | with(.[] ; .value=(.value | to_entries))' -
+      expected: expected/metrics-has-value-label.yml
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_nginx_service_http_requests_increment --service-name=nginx::e2e-test | yq e 'to_entries' -
+      expected: expected/metrics-has-value.yml
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics multiple-linear --name=meter_nginx_service_error_log_count --service-name=nginx::e2e-test | yq e 'to_entries | with(.[] ; .value=(.value | to_entries))' -
+      expected: expected/metrics-has-value-label.yml
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql logs list --service-name=nginx::e2e-test
+      expected: expected/logs.yml
+
```
Comment (wankai123) [inline test/e2e-v2/cases/nginx/e2e.yaml:65]:
```suggestion
```

File: test/e2e-v2/cases/nginx/e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  retry:
+    # max retry count
+    count: 20
+    # the interval between two retries, in millisecond.
+    interval: 3s
+  cases:
+    # service cases
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql service ls
+      expected: expected/service.yml
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_nginx_service_http_requests --service-name=nginx::e2e-test | yq e 'to_entries' -
```
Comment (wankai123) [inline test/e2e-v2/cases/nginx/e2e.yaml:49]: We should use MQE for metrics query

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            "TIME_SERIES_VALUES"
+          ],
+          "graph": {
+            "type": "Area",
+            "opacity": 0.4,
+            "showXAxis": true,
+            "showYAxis": true
+          },
+          "widget": {
+            "title": "HTTP Status 4xx Percent (%)"
```
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json:131]: Is this a Percentage Per Minute?

File: test/e2e-v2/cases/nginx/e2e.yaml (hunk 3 above)
Comment (weixiang1862) [reply test/e2e-v2/cases/nginx/e2e.yaml:49]: Ok, thanks.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json (hunk 4 above)
Comment (weixiang1862) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json:131]: Yes.

**Comment by wu-sheng** [comment]
> error.log has log level, accesslog doesn't. Should I set the default log level for accesslog?

Which logs are you gathering from accesslog? The log output should include the level, isn't it?

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json (hunk 4 above)
Comment (wankai123) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json:131]: Then the title `HTTP Status 4xx Percent Per Minute (%)` is better?

**Comment by weixiang1862** [comment]
> > error.log has log level, accesslog doesn't. Should I set the default log level for accesslog?
> 
> Which logs are you gathering from accesslog? The log output should include the level, isn't it?

nginx output accesslog and errorlog in two files, I collect both of them. error log has log level but access log doesn't.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json (hunk 4 above)
Comment (weixiang1862) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/nginx/nginx-endpoint.json:131]: Ok, I will change it.

**Comment by wu-sheng** [comment]
```
log_format myformat '$remote_addr - $remote_user [$time_local] '
                    '"$request" $status $body_bytes_sent '
                    '"$http_referer" "$http_user_agent" '
                    '"$http_x_forwarded_for"';

access_log /path/to/access.log myformat;
```

I think we should extract `$status` from a defined format. And tag the error `level` and  `http.status_code`. Also, add `http.status_code` into `searchableLogsTags`.

**Comment by weixiang1862** [comment]
> ```
> log_format myformat '$remote_addr - $remote_user [$time_local] '
>                     '"$request" $status $body_bytes_sent '
>                     '"$http_referer" "$http_user_agent" '
>                     '"$http_x_forwarded_for"';
> 
> access_log /path/to/access.log myformat;
> ```
> 
> I think we should extract `$status` from a defined format. And tag the error `level` and  `http.status_code`. Also, add `http.status_code` into `searchableLogsTags`.

ok, i will extract status and level tag.

**Comment by weixiang1862** [comment]
accesslog http.status_code & errorlog level is searchable:

ae8090779732ffa4a48bdff2fee92b4
5aa9fe469fa5535fe665772d6b00acc

File: docs/en/setup/backend/backend-nginx-monitoring.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
+# Nginx monitoring
+## Nginx performance from nginx-lua-prometheus
+The [nginx-lua-prometheus](https://github.com/openresty/lua-nginx-module) is a lua library that can be used with Nginx to collect metrics
```
Comment (spacewander) [inline docs/en/setup/backend/backend-nginx-monitoring.md:3]: The link is wrong?

Comment (weixiang1862) [reply docs/en/setup/backend/backend-nginx-monitoring.md:3]: Yes, thanks. I will fix this.

**Review by wankai123 — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
@weixiang1862 Generally, this should be good. I will wait for others' review. And would you be able to add Nginx monitoring

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
And I believe another blog to introduce your new thing would be cool. This should be more powerful than you introduced in the last one. WDYT?

**Comment by weixiang1862** [comment]
> And I believe another blog to introduce your new thing would be cool. This should be more powerful than you introduced in the last one. WDYT?

Yes, I will add a new blog to introduce it and add it to showcase, I think they will be ok in the end of this month.

**Comment by wu-sheng** [comment]
> > And I believe another blog to introduce your new thing would be cool. This should be more powerful than you introduced in the last one. WDYT?
> 
> Yes, I will add a new blog to introduce it and add it to showcase, I think they will be ok in the end of this month.

The time is totally flexible, and up to your reality.

**Pull request merged**